daemon/cmd/local_node_sync.go

```diff
@@ -120,6 +122,28 @@ func (ini *localNodeSynchronizer) getK8sLocalNode(ctx context.Context) (*slim_co
 	return nil, ctx.Err()
 }
 
+// getK8sLocalCiliumNode returns the CiliumNode object for the local node if it exists at the type
+// of the call.
+// In the case that the resource event is synced without a ciliumnode upsert event, we return nil.
+func (ini *localNodeSynchronizer) getK8sLocalCiliumNode(ctx context.Context) *v2.CiliumNode {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	select {
+	case <-ctx.Done():
+		return nil
```

Comment:
Nit. This explicit check is not necessary because the events channel is closed when the context is canceled.